tests/Unit/Proposition/PropositionManagerTest.php

```diff
@@ -0,0 +1,184 @@
+
+    $platformValue = PlatformFacade::get($key);
+    $propositionValue = PropositionFacade::get($key);
+
+    expect($propositionValue)->toBe($platformValue);
+})->with([
+    ['name', Platform::MYPARCEL_NAME],
+    ['human', Platform::MYPARCEL_NAME],
+    ['localCountry', Platform::MYPARCEL_NAME],
+    ['defaultCarrier', Platform::MYPARCEL_NAME],
+    ['name', Platform::SENDMYPARCEL_NAME],
+    ['human', Platform::SENDMYPARCEL_NAME],
+    ['localCountry', Platform::SENDMYPARCEL_NAME],
+]);
+
+it('returns same carriers as Platform facade', function (string $platform) {
+    MockPdkFactory::create(['platform' => $platform]);
+
+    $platformCarriers = PlatformFacade::getCarriers();
+    $propositionCarriers = PropositionFacade::getCarriers();
+
+    expect($propositionCarriers->toArray())->toBe($platformCarriers->toArray());
+})->with('platforms');
+
+/**
+ * CONSTANTS TESTS
+ * 
+ * Ensure Proposition constants are defined correctly
+ */
+
+it('has correct proposition constants', function () {
+    expect(Proposition::MYPARCEL_ID)->toBe(1)
+        ->and(Proposition::MYPARCEL_NAME)->toBe('myparcel')
+        ->and(Proposition::SENDMYPARCEL_ID)->toBe(2)
+        ->and(Proposition::SENDMYPARCEL_NAME)->toBe('belgie');
+});
+
+it('proposition constants match platform constants', function () {
+    expect(Proposition::MYPARCEL_ID)->toBe(Platform::MYPARCEL_ID)
+        ->and(Proposition::MYPARCEL_NAME)->toBe(Platform::MYPARCEL_NAME)
+        ->and(Proposition::SENDMYPARCEL_ID)->toBe(Platform::SENDMYPARCEL_ID)
+        ->and(Proposition::SENDMYPARCEL_NAME)->toBe(Platform::SENDMYPARCEL_NAME);
+});
+
+/**
+ * TRANSITION SUPPORT TESTS
+ * 
+ * Test support for both platformId 2 and 3 during transition
+ */
+
+it('supports both platformId 2 and 3 for SendMyParcel', function (int $platformId) {
+    // Test the PropositionManager logic directly
+    // The manager checks for both Platform::SENDMYPARCEL_ID (2) and hardcoded 3
+    
+    // For platformId 2 (current constant)
+    if ($platformId === 2) {
+        expect($platformId)->toBe(Platform::SENDMYPARCEL_ID);
+    }
+    
+    // For platformId 3 (future/transition)
+    if ($platformId === 3) {
+        expect($platformId)->toBe(3);
+    }
+    
+    // Both should be supported according to PropositionManager logic
+    expect($platformId === Platform::SENDMYPARCEL_ID || $platformId === 3)->toBeTrue();
+})->with([
+    'current ID' => [2],
+    'future ID' => [3],
+]);
+
+it('handles missing account settings gracefully', function () {
+    // Create PDK without account settings to trigger catch block
+    MockPdkFactory::create([
+        'platform' => null,
+        'account' => null,
+    ]);
+    
+    $propositionName = PropositionFacade::getPropositionName();
+    
+    // Should default to MyParcel when account settings are unavailable
+    expect($propositionName)->toBe(Platform::MYPARCEL_NAME);
+});
+
+it('uses configured platform over account settings', function () {
```

Comment:
Check even of dit echt moet, volgens mij niet namelijk. Ik denk dat het eerder tot problemen kan leiden (platform dat niet klopt met de api key bv)